qrisk/tests/test_stats.py

```diff
@@ -283,21 +484,121 @@ def test_sortino(self, returns, required_return, period, expected):
                     expected[i],
                     DECIMAL_PLACES)
 
+    # A large Sortino ratio indicates there is a low probability of a large
+    # loss, therefore randomly changing values larger than required return to a
+    # loss of 25 percent decreases the ratio.
+    @parameterized.expand([
+        (noise_uniform, 0),
+        (noise, 0),
+    ])
+    def test_sortino_add_noise(self, returns, required_return):
+        sr_1 = qrisk.sortino_ratio(returns, required_return)
+        upside_values = returns[returns > required_return].index.tolist()
+        # Add large losses at random upside locations
+        loss_loc = random.sample(upside_values, 2)
+        returns[loss_loc[0]] = -0.01
+        sr_2 = qrisk.sortino_ratio(returns, required_return)
+        returns[loss_loc[1]] = -0.01
+        sr_3 = qrisk.sortino_ratio(returns, required_return)
+        assert sr_1 > sr_2
+        assert sr_2 > sr_3
+
+    # Similarly, randomly increasing some values below the required return to
+    # the required return increases the ratio.
+    @parameterized.expand([
+        (noise_uniform, 0),
+        (noise, 0)
+    ])
+    def test_sortino_sub_noise(self, returns, required_return):
+        sr_1 = qrisk.sortino_ratio(returns, required_return)
+        downside_values = returns[returns < required_return].index.tolist()
+        # Replace some values below the required return to the required return
+        loss_loc = random.sample(downside_values, 2)
+        returns[loss_loc[0]] = required_return
+        sr_2 = qrisk.sortino_ratio(returns, required_return)
+        returns[loss_loc[1]] = required_return
+        sr_3 = qrisk.sortino_ratio(returns, required_return)
+        assert sr_1 <= sr_2
+        assert sr_2 <= sr_3
+
+    # Translating the returns and required returns by the same amount
+    # should not change the sortino ratio.
+    @parameterized.expand([
```

Comment:
Should we test that if we change it by different amounts sortino ratio does change?